subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/KotlinPluginSmokeTest.groovy

```diff
@@ -221,6 +221,7 @@ class KotlinPluginSmokeTest extends AbstractPluginValidatingSmokeTest implements
         }
         if (isAndroidKotlinPlugin(testedPluginId)) {
             AGP_VERSIONS.assumeCurrentJavaVersionIsSupportedBy(androidVersion)
+            AGP_VERSIONS.assumeAgpSupportsKotlinVersion(androidVersion, version)
```

Comment:
Would it be better to have a single `assume` method? It seems like when we call one we'll always want to also call the other, which makes this redundant and error-prone.

Reply:
I added another assume method, which checks the Java and the Kotlin version.